features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Create/Default/APICreateDefault.jsx

```diff
@@ -198,45 +201,112 @@ function APICreateDefault(props) {
-                if (workflowStatus === APICreateDefault.WORKFLOW_STATUS.CREATED) {
-                    Alert.info(intl.formatMessage({
-                        id: 'Apis.Create.Default.APICreateDefault.success.publishStatus',
-                        defaultMessage: 'Lifecycle state change request has been sent',
-                    }));
-                } else {
-                    Alert.info(intl.formatMessage({
-                        id: 'Apis.Create.Default.APICreateDefault.success.otherStatus',
-                        defaultMessage: 'API updated successfully',
-                    }));
-                }
-                history.push(`/apis/${api.id}/overview`);
-            })
-            .catch((error) => {
-                if (error.response) {
-                    Alert.error(error.response.body.description);
-                    setPageError(error.response.body);
-                } else {
-                    const message = 'Something went wrong while publishing the API';
-                    Alert.error(intl.formatMessage({
-                        id: 'Apis.Create.Default.APICreateDefault.error.errorMessage',
-                        defaultMessage: message,
-                    }));
-                    setPageError(message);
-                }
-                console.error(error);
-            })
-            .finally(() => {
-                setIsPublishing(false);
-            }));
+        const restApi = new API();
+        setIsPublishButtonClicked(true);
+        createAPI().then((api) => {
+            setIsRevisioning(true);
+            const body = {
+                description: 'Initial Revision',
+            };
+            restApi.createRevision(api.id, body)
+                .then((api1) => {
+                    const revisionId = api1.body.id;
+                    Alert.info('API Revision created successfully');
+                    setIsRevisioning(false);
+                    const envList = settings.environment.map((env) => env.name);
+                    const body1 = [];
+                    if (envList.length > 0) {
+                        body1.push({
+                            name: envList[0],
+                            displayOnDevportal: true,
+                        });
+                    }
+                    // for (let i = 0; i < envList.length; i++) {
+                    //     body1.push({
+                    //         name: envList[i],
+                    //         displayOnDevportal: true,
+                    //     });
+                    // }
+                    setIsDeploying(true);
+                    restApi.deployRevision(api.id, revisionId, body1)
+                        .then(() => {
+                            Alert.info('API Revision Deployed Successfully');
+                            setIsDeploying(false);
+                        })
+                        .catch((error) => {
+                            if (error.response) {
```

Comment:
In case of an error shouldn't we stop proceeding forward ?

![image](https://user-images.githubusercontent.com/3313885/105876059-93d17900-6024-11eb-93b4-b6b6e788f6cb.png)

I got above error when

Reply:
When this happens I can't see the revisions diagram (circular thing) on the Deployments page

![image](https://user-images.githubusercontent.com/3313885/105878457-15c2a180-6027-11eb-9869-14d66c1e15cd.png)